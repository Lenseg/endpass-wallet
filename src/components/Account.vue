<template>
  <div class="columns">
    <div class="column">
      <p class="heading">Ethereum Address</p>
      <div class="address">
        <div class="identicon" :style="{ backgroundImage: 'url(' + icon + ')' }"></div>
        <p class="code">
          {{ address }}
        </p>
      </div>
    </div>
    <div class="column">
      <balance :amount="balance" :currency="currency" />
    </div>
  </div>
</template>

<script>
import blockies from 'ethereum-blockies';
import Balance from '@/components/Balance';

export default {
  name: 'Account',
  components: { Balance },
  props: {
    address: {
      type: String,
      required: true,
    },
    currency: {
      type: String,
      default: 'ETH'
    },
    balance: {
      type: String,
      default: '0',
    },
  },
  computed: {
    icon() {
      const seed = this.address.toLowerCase();
      return blockies
        .create({ seed:seed, size: 8, scale: 16})
        .toDataURL();
    },
  },
};
</script>

<style>
.identicon {
  display: inline-block;
  width: 2rem;
  height: 2rem;
  background-size:cover;
  background-repeat: no-repeat;
  border-radius: 50%;
}
</style>
