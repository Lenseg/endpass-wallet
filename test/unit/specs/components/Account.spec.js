import { shallow } from '@vue/test-utils';

import Account from '@/components/Account';

describe('Account', () => {
  describe('render', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(Account, {
        propsData: {
          address: '0x9999999999999999999999999999999999999999',
        },
      });
    });

    it('should be a Vue component', () => {
      expect(wrapper.name()).toBe('Account');
      expect(wrapper.isVueInstance()).toBeTruthy();
    });

    it('should render component', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should render props', () => {
      const props = {
        address: '0x9eceefdf3554e178a6549006f2c02163e63c9fd8',
      };

      wrapper.setProps(props);

      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
