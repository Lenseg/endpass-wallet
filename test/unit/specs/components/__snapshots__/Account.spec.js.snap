// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account render should render component 1`] = `
<div
  class="columns"
>
  <div
    class="column"
  >
    <p
      class="heading"
    >
      Ethereum Address
    </p>
     
    <p
      class="code address"
    >
      <img
        src=""
      />
       0x9999999999999999999999999999999999999999
    
    </p>
  </div>
   
  <div
    class="column"
  >
    <!---->
  </div>
</div>
`;

exports[`Account render should render props 1`] = `
<div
  class="columns"
>
  <div
    class="column"
  >
    <p
      class="heading"
    >
      Ethereum Address
    </p>
     
    <p
      class="code address"
    >
      <img
        src=""
      />
       0x9eceefdf3554e178a6549006f2c02163e63c9fd8
    
    </p>
  </div>
   
  <div
    class="column"
  >
    <!---->
  </div>
</div>
`;
